pkg/logql/error.go

```diff
@@ -64,8 +70,22 @@ func (e pipelineError) Error() string {
 		e.errorType, e.metric, e.errorType)
 }
 
-// IsPipelineError tells if the error is generated by a Pipeline.
-func IsPipelineError(err error) bool {
-	_, ok := err.(*pipelineError)
-	return ok
+// Is allows to use errors.Is(err,ErrPipeline) on this error.
+func (e pipelineError) Is(target error) bool {
```

Comment:
@kavirajk this was due from one of your comment in previous PR. :)

Reply:
but we don't have to implement `Is` for the custom error type to use `errors.Is` :)

Reply:
we do if the error is not a static object which is the case.